### PR TITLE
test: Add Zowe test suite

### PR DESCRIPTION
**Pull request by nalmabrcom**
## How to test
You need to mount the `zowe` folder
```
che-che4z-lsp-for-cobol/tests/test_files/zowe:/home/theia/.zowe
```
Then simply run the suite
```
npm run cy:run -- --config baseUrl=https://theia --spec cypress/integration/LSP/zowe.spec.js -b chrome --headless
```

## Results
```
  This is a Zowe integration spec file
    TC219568 Copybook is (semi) automatically downloaded
      √ Checks that copybook was downloaded (9384ms)
    TC219569 Nested Copybook is (semi) automatically downloaded
      √ Checks that nested copybooks were downloaded (15042ms)
    TC219570 Nested Copybook is (semi) automatically downloaded
      √ Cobol file has a syntax error in case copybook is missing  (18179ms)
    TC219571 Cobol file has a syntax error in case nested copybook is missing
      √ Checks nested copybook missing (8914ms)
    TC312870 Wrong profile credentials - one profile
      √ Wrong profile credentials - one profile (13802ms)
    TC312876 Wrong profile credentials - no password
      √ Verify that server detects a no Pass credential response. (12891ms)
    TC312871 Wrong profile credentials - change to a valid profile
      √ Wrong profile credentials - change to a valid profile (10932ms)
    TC312877 Wrong profile credentials - change from invalid  profile to noPass and vice versa
      √ Wrong profile credentials - change from invalid  profile to noPass and vice versa (16152ms)
    TC320158 Get 501 error
      √ Get 501 error (13043ms)
    TC219573 Nested Recursive Copybooks are downloaded and error is shown
      √ Nested Recursive Copybooks are downloaded and error is shown (18926ms)


  10 passing (4m)
```

Signed-off-by: Maryna Nalbandian <maryna.nalbandian@broadcom.com>

### CODE REVIEW

**Review by temanbrcom — APPROVED** [review]
What about storing the descriptions as a wiki page?

**Review by asatklichov — APPROVED** [review]
🦅

File: tests/cypress/integration/LSP/zowe.spec.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  const NOTFOUND = basePath + 'NOTFOUND.cpy';
+  const NOTFOUND_NESTED = basePath + 'COPYNF.cpy';
+  const ERR = basePath + 'ERR';
+
+  /* Functions Panel */
+  function getHoverMessage(line, message) {
+    return cy.goToLine(line).getCurrentLineErrors({ expectedLine: line }).getHoverErrorMessage().contains(message);
+  }
+
+  function addCopybookAndgetHoverMessage(line, copybook, message) {
```
Comment (grianbrcom) [inline tests/cypress/integration/LSP/zowe.spec.js:35]: addCopybookAnd**G**etHoverMessage

**Review by grianbrcom — APPROVED** [review]
:beetle:

File: tests/cypress/integration/LSP/zowe.spec.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  const ASTRO = basePath + 'ASTRO.cpy';
+  const FOO = basePath + 'FOO.cpy';
+  const NOTFOUND = basePath + 'NOTFOUND.cpy';
+  const NOTFOUND_NESTED = basePath + 'COPYNF.cpy';
+  const ERR = basePath + 'ERR';
+
+  /* Functions Panel */
+  function getHoverMessage(line, message) {
+    return cy.goToLine(line).getCurrentLineErrors({ expectedLine: line }).getHoverErrorMessage().contains(message);
+  }
```
[the remark this answers is not in the document]
Comment (nalmabrcom) [reply tests/cypress/integration/LSP/zowe.spec.js:33]: It works in 2 ways of declarations: JS functions and Cypress functions.
In this case, I think we can use JS functions and declare them as commands in `support/*.ts` if needed.

**Comment by nalmabrcom** [comment]
> What about storing the descriptions as a wiki page?

Might be a good idea!

**Pull request merged**